lib/puppet/util/servicenow.rb

```diff
@@ -89,4 +89,54 @@ def do_snow_request(uri, http_verb, body, user: nil, password: nil, oauth_token:
     end
   end
   module_function :do_snow_request
+
+  def requested_status?(status, corrective_change, noop_pending)
+    change_types = settings['incident_creation_report_attributes']
+
+    if change_types.include? 'none'
+      Puppet.info('servicenow reporting: incident_creation_report_attributes includes \'none\'.')
+      return false
+    elsif change_types.include? 'unchanged'
+      Puppet.info('servicenow reporting: incident_creation_report_attributes includes \'unchanged\'.')
+      return true
+    elsif change_types.include? 'all'
+      change_types = ['failed_changes', 'corrective_changes', 'intentional_changes', 'pending_changes']
+    end
+
+    report_on_noop = change_types.include? 'pending_changes'
```

Comment:
That's fine. I'm probably gonna look into at least getting the intentional changes bit working (I don't think it's worth waiting for Puppet to get it in because that will take a while) so can do that once this is merged.

Reply:
Meh nevermind, given that we also want `pending_corrective_changes` and `pending_intentional_changes` I'll wait for the Puppet-side part to add those fields.